packages/expo-file-system/ios/FileSystemModule.swift

```diff
@@ -77,6 +77,7 @@ public final class FileSystemModule: Module {
       guard url.isFileURL else {
         throw InvalidFileUrlException(url)
       }
+      try ensurePathPermission(appContext, path: url.appendingPathComponent("..").path, flag: .write)
```

Comment:
In `deleteAsync` we should check the permission on the parent directory (it regressed during migration to Swift)